app/controllers/histories_controller.rb

```diff
@@ -8,9 +8,45 @@ class HistoriesController < ApplicationController
   def create
     redirect_to root_url unless current_user.can_create_subject_history?
 
-    History.create!(patient_id: params.permit(:patient_id)[:patient_id],
+    history = History.new(patient_id: params.permit(:patient_id)[:patient_id],
+                          created_by: current_user.email,
+                          comment: params.permit(:comment)[:comment],
+                          history_type: params.permit(:type)[:type] || 'Comment')
+
+    history.original_comment = history
+
+    # Attempt to save and continue; else if failed redirect to index
+    render(json: history.errors, status: 422) && return unless history.save
+
+    render(json: history) && return
+  end
+
+  # "Edits" a history comment - a new history comment is created with the updated comment text and a reference to the id of the original
+  def edit
+    redirect_to root_url unless current_user.can_create_subject_history?
+
+    patient = current_user.viewable_patients.find_by(id: params.permit(:patient_id)[:patient_id])
+    history = patient.histories.find_by(id: params.permit(:id)[:id])
+    redirect_to root_url && return if history.nil? || history.history_type != 'Comment' || history.created_by != current_user.email
+
+    History.create!(patient_id: history.patient_id,
                     created_by: current_user.email,
                     comment: params.permit(:comment)[:comment],
-                    history_type: params.permit(:type)[:type] || 'Comment')
+                    history_type: params.permit(:type)[:type] || 'Comment',
+                    original_comment_id: history.original_comment_id)
+  end
+
+  # "Deletes" a history comment - does not actually remove the comment from the database
+  # but adds a deleted_by and deleted_reason that show the comment was deleted
+  def delete
```

Comment:
this is probably not a bad time to write a `histories_controller_test.rb` file

Reply:
would be nice to have more test cases, but here's a start: https://github.com/SaraAlert/SaraAlert/pull/793/commits/c8f1f62444e06ae8640e77e29be166045165d6dd#diff-e24e46e4cebd169bcec6422cfdd877e9bbfa4b402096d36398418837e351a3d5R5

Reply:
I fleshed these out a bit more here: https://github.com/SaraAlert/SaraAlert/pull/793/commits/37e5e22912de96896e1da829ddb1db84305d0b37. I think it's sufficient for now